packages/manager/apps/pci-databases-analytics/src/pages/services/[serviceId]/indexPatterns/_components/IndexesTableColumns.component.tsx

```diff
@@ -0,0 +1,141 @@
+import { useTranslation } from 'react-i18next';
+import { ColumnDef } from '@tanstack/react-table';
+
+import { MoreHorizontal } from 'lucide-react';
+import {
+  Tooltip,
+  TooltipProvider,
+  TooltipTrigger,
+} from '@/components/ui/tooltip';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from '@/components/ui/dropdown-menu';
+import { Button } from '@/components/ui/button';
+
+import * as database from '@/types/cloud/project/database';
+import { useServiceData } from '../../Service.context';
+import FormattedDate from '@/components/formatted-date/FormattedDate.component';
+import { DatatableSortableHeader } from '@/components/data-table/DatatableSortableHeader.component';
+
+interface IndexesTableColumnsProps {
+  onDeleteClick: (index: database.opensearch.Index) => void;
+}
+export const getIndexesColumns = ({
```

Comment:
AFAIS, this function does not return neither a React Component or a hook, so you cannot invoke hooks inside because it violates the rule of hooks ;)
Only hooks or React component can use hooks :)

Reply:
As for `PatternTableColumns` component, I'm a bit confused with the naming